tests/components/minecraft_server/test_config_flow.py

```diff
@@ -56,9 +63,26 @@ async def test_connection_failed(hass: HomeAssistant) -> None:
         assert result["errors"] == {"base": "cannot_connect"}
 
 
-async def test_connection_succeeded(hass: HomeAssistant) -> None:
+async def test_bedrock_connection_failed(hass: HomeAssistant) -> None:
```

Comment:
In each test I already check for `FORM` or `CREATE_ENTRY`. Do you mean the order of the asserts? Should the check for flow result be always the last assert in the test function? Or do you mean something else?

Reply:
Sorry, I'm a bit lost here. I checked a few other integrations and didn't see much of a difference. The config flow of Minecraft Server either is successful (`CREATE_ENTRY`) or it cannot connect. In that case the same `FORM` is shown only with a difference, that the error message is shown additionally. In that form the user can correct the user input and retry.

Reply:
It's something that we like new tests to have, to avoid code that could stick around and put the config flow in some kind of lock. So the trick is to finish the FORM, so it will become either a CREATE_ENTRY or an ABORT

Reply:
My problem is, that I don't abort my config flow, therefore (if I understood correctly) I can never reach the ABORT state. Let me try to make this more clear with screenshots.

This is the initial form. I entered an invalid input and after confirming we come to the second screenshot.
![Bildschirmfoto vom 2023-09-27 10-19-02](https://github.com/home-assistant/core/assets/1382097/15b886c8-a35c-4103-8894-1a6ab44a399c)

After the config flow detected a connection error, the same form is shown with additionally the error message.
![Bildschirmfoto vom 2023-09-27 10-19-40](https://github.com/home-assistant/core/assets/1382097/646d04a1-2385-427a-9cbd-727de62a1cd1)

So the config flow doesn't abort the flow, even if a connection error is detected.

Reply:
Understood, but then we would like the tests to first reach the state where we show the user an error, and then retry again with valid info, making it succeed. This is to make sure the config flow doesn't end up in an unsolvable.

Reply:
This is something I can add, no problem. Thanks :+1:

Reply:
hah, explaining isn't my strongest point, but we got there in the end :)

Reply:
Done, added a new test case for config flow recovery (successful connection after a failed one).